pkg/controller/knativeserving/knativeserving_controller.go

```diff
@@ -228,6 +230,25 @@ func (r *ReconcileKnativeServing) checkDeployments(instance *servingv1alpha1.Kna
 	return nil
 }
 
+// Check for all dependencies
+func (r *ReconcileKnativeServing) checkDependencies(instance *servingv1alpha1.KnativeServing) error {
+	defer r.updateStatus(instance)
+
+	istio := schema.GroupVersionKind{Group: "networking.istio.io", Version: "v1alpha3", Kind: "virtualservice"}
+	list := &unstructured.UnstructuredList{}
+	list.SetGroupVersionKind(istio)
+	if err := r.client.List(context.TODO(), nil, list); err != nil {
+		msg := fmt.Sprintf("Istio not detected")
```

Comment:
In the interest of usability: Should we call out "ServiceMesh not detected" here? I know that technically we only need Istio, but the user of Openshift Serverless might think "hey, what the hell is Istio? I only see something like ServiceMesh referenced in the docs". Maybe we could also write something like "Istio not detected, ServiceMesh is likely not installed" to satisfy both ends?

Reply:
Good call. Done.